Paralayout/PixelRounding.swift

```diff
@@ -34,7 +34,17 @@ extension UIScreen: ScaleFactorProviding {
 extension UIView: ScaleFactorProviding {
 
     public var pixelsPerPoint: CGFloat {
+        let scaleFromTraits = traitCollection.displayScale
+
+        // The trait collection is the authority for display scale, but sometimes the trait collection does not have a
+        // value for the scale, in which case it will return 0, breaking all the things.
+        if scaleFromTraits > 0 {
+            return scaleFromTraits
+        }
+
+        #if os(iOS)
         return (window?.screen ?? UIScreen.main).pixelsPerPoint
+        #endif
```

Comment:
This OS check is a precursor to #111, where we can't use `UIScreen`

Reply:
We might also be able to fall back to `window?.traitCollection ?? UITraitCollection.current` as an alternative way to get the scale instead of `UIScreen`

Reply:
If it's in a window, it _should_ get the traits from the window automatically, so I don't think the first part is necessary. Falling back to the current global trait collection is a really interesting idea though. I think in practice it would probably be safe to do so, but `UITraitCollection.current` has some really weird behavior around when it's valid to read, so I'm a bit hesitant to use it here. I think for now using `UIScreen` is safer (and should give the same value).